client/state/sites/domains/test/reducer.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import { domains } from '../reducer';
+import {
+	SITE_DOMAINS_FETCH,
+	SITE_DOMAINS_FETCH_COMPLETED,
+	SITE_DOMAINS_FETCH_FAILED,
+	SERIALIZE,
+	DESERIALIZE
+} from 'state/action-types';
+
+describe( 'reducer', () => {
+	describe( '#domains()', () => {
+		it( 'should return an empty state when original state is undefined and action is empty', () => {
+			const state = domains( undefined, {} );
+
+			expect( state ).to.eql( {} );
+		} );
+
+		it( 'should return an empty state when original state and action are empty', () => {
+			const original = Object.freeze( {} );
+			const state = domains( original, {} );
+
+			expect( state ).to.eql( original );
+		} );
+
+		it( 'should return an empty state when original state is undefined and action is unknown', () => {
+			const state = domains( undefined, {
+				type: 'SAY_HELLO',
+				siteId: 11111111
+			} );
+
+			expect( state ).to.eql( {} );
+		} );
+
+		it( 'should return the original state when action is unknown', () => {
+			const original = Object.freeze( {
```

Comment:
Let's update usages of Object.freeze to `deepFreeze`, since Object.freeze is shallow. This is available via: `import deepFreeze from 'deep-freeze';`